### PR TITLE
create api routes

### LINKED ISSUE #14: create back-end routes

### PR DESCRIPTION
**Pull request by theo333**
Closes #14 

Created routes for 
- challenges
- userchallenges
- solutions
- users
- (not for) images

### CODE REVIEW

File: server/routes/api/challenges.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const router = require('express').Router();
+const { Challenge } = require('../../db');
+
+/**  /api/challenges **/
+
+// get all challenges
+router.get('/', (req, res, next) => {
+  console.log('get all challenges route');
```
Comment (hgranthorner) [inline server/routes/api/challenges.js]: let's remove these console.logs

File: server/routes/api/solutions.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  Solution.findByPk(req.params.id)
+    .then(solution => solution.update(req.body))
+    .then(updatedSolution => res.send(updatedSolution))
+    .catch(next);
+});
+
+// delete a challenge solution
+router.delete('/:id', (req, res, next) => {
+  Solution.findByPk(req.params.id)
+    .then(solution => solution.destroy())
```
Comment (hgranthorner) [inline server/routes/api/solutions.js]: do we need to follow this up with solution.save()?

File: server/routes/api/challenges.js (hunk 1 above)
Comment (theo333) [reply server/routes/api/challenges.js]: Removed in the PR just submitted for Image model

File: server/routes/api/solutions.js (hunk 2 above)
Comment (theo333) [reply server/routes/api/solutions.js]: No.

**Pull request merged**